### PR TITLE
Some refactoring and UI fixes.

### PR DESCRIPTION
**Pull request by N3da**

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -17,43 +17,45 @@
...
-        body = document.getElementsByTagName('body')[0],
+        randomElement = $('#random'),
+        uploadElement = $('#upload'),
+        firstPageElements = $('.-level1-'),
+        secondPageElements = $('.-level2-'),
         arrayKeyPattern = /(\d+)/g,
-        queryPattern = /[?&#]+([\w]+)=([^&#]*)/g,
         reservedKeysPattern = /-[a-z]+-/g,
-        whitespacePattern = /[\s]+/g,
+        whiteSpacePattern = /[\s+]/g,
```
Comment (mtth) [inline js/index.js]: Did you mean `/\s+/g`? (Or is the `+` supposed to match too?)

File: js/index.js [hunk 2]
```diff
@@ -230,71 +254,59 @@
...
-      }
     }
     
     /**
     * Will save cursor position inside element `elemId` before running callback function f,
     * and restores it after f is finished. 
     */ 
-    function runPreservingCursorPosition(elementId, f) {
+    function runPreservingCursorPosition(elementId, f, options) {
+      var context = options ? options.context: undefined;
```
Comment (mtth) [inline js/index.js]: Unless you need the context to be exactly `undefined`, this is often rewritten as `options && options.context`. (Same below.)

File: todo.txt [hunk 3]
```diff
@@ -1,6 +1,2 @@
-Support reverse highlighting.
-Align/fix the error message bar.
-Fix decoding
-colors
-Fix adding spaces in schema field names.
-Refactor how error message is handled.
+minify stuff.
```
Comment (mtth) [inline todo.txt]: Did you mean to check this file in? (Also I thought this one was done?)

**Comment by mtth** [comment]
Thanks! Just a few minor comments inline.

File: js/index.js (hunk 1 above)
Comment (N3da) [inline js/index.js]: No you are right. I added `,` too but then I removed it and forgot the []s.

File: todo.txt (hunk 3 above)
Comment (N3da) [inline todo.txt]: No I did not :(

**Comment by mtth** [comment]
Thank you :).

**Pull request merged**